packages/wonder-blocks-form/src/components/labeled-text-field.js

```diff
@@ -53,25 +97,42 @@ export default class LabeledTextField extends React.Component<Props, State> {
         });
     };
 
+    handleOnChange: (newValue: string) => mixed = (newValue) => {
+        this.setState({
+            value: newValue,
+        });
+    };
+
     render(): React.Node {
-        const {disabled} = this.props;
+        const {id, type, label, description, disabled} = this.props;
+
         return (
-            <FieldHeading
-                field={
-                    <TextField
-                        id="tf-1"
-                        value="Value"
-                        placeholder="Placeholder"
-                        disabled={disabled}
-                        onChange={() => {}}
-                        onFocus={this.handleOnFocus}
-                        onBlur={this.handleOnBlur}
+            <IDProvider id={id} scope="field">
+                {(uniqueId) => (
+                    <FieldHeading
+                        id={uniqueId}
+                        field={
+                            <TextField
+                                id={`${uniqueId}-field`}
+                                aria-describedby={`${uniqueId}-error`}
+                                aria-invalid={
+                                    this.state.error ? "true" : "false"
+                                }
+                                type={type}
+                                value={this.state.value}
+                                placeholder="Placeholder"
+                                disabled={disabled}
+                                onChange={this.handleOnChange}
+                                onFocus={this.handleOnFocus}
+                                onBlur={this.handleOnBlur}
```

Comment:
style nit: usually these are called `handleChange`, `handleBlur`, etc.

Reply:
Just to be sure... you are referring to the handle function, right? Not the prop.

So I would just remove the "On" in the function name.
```js
this.handleOnChange
```
to
```js
this.handleChange
```